### ISSUE TITLE
Property 'redraw' does not exist on type 'Event'

### ISSUE THREAD

**Issue by pygy** [spacejack/mithril.d.ts#17]
Hey there :-)

It may be nice to have maybe a `Mithril.Event` with an added `redraw: boolean` field.

Or did I miss it?

**Comment by spacejack** [comment]
Yeah that would be handy as a convenience type to include. I think you'd still need to annotate it yourself:

```typescript
m('button', {onclick: (e: m.Event) => {e.redraw = false}}, "Don't redraw!");
```

Unless we can do something clever with the `Attrs` type...

**Comment by spacejack** [comment]
Thinking about this a bit more, might be more useful to extend all possible events. It could be made more user friendly by adding [these](https://github.com/Microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L3579-L3605) and [these](https://github.com/Microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L4537-L4603) (and other...?) properties to `Attrs`.

Eg:

```typescript
export interface VEvent {
    redraw: boolean
}

export interface Attrs {
    // ...
    onblur?: (this: HTMLElement, ev: FocusEvent & VEvent) => any;
    onclick?: (this: HTMLElement, ev: MouseEvent & VEvent) => any;
    // etc...
```

**Comment by spacejack** [comment]
Here's a gist of everything I can think of at the moment:

https://gist.github.com/spacejack/73ae839c4a80a2dfaa88294caec3605f

**Comment by spacejack** [comment]
I've added a new branch `event-types`.

@isiahmeadows, @andraaspar what do you think: https://github.com/spacejack/mithril.d.ts/compare/event-types

**Comment by pygy** [comment]
LGTM :-)

**Comment by dead-claudia** [comment]
LGTM, although I wish we could concatentate string keys with mapped types, so we wouldn't need to manage the big list ourselves... (TypeScript doesn't allow it though, sadly.)

**Comment by spacejack** [comment]
Something I should point out about this change, if you're using external event handler functions, you'd need to annotate them like this: https://github.com/spacejack/mithril.d.ts/blob/event-types/test/test-api.ts#L649

**Comment by oldrich-s** [comment]
@isiahmeadows Is it not solved by my proposal?: https://github.com/spacejack/mithril.d.ts/issues/24
